src/dump_load.cc

```diff
@@ -309,6 +317,9 @@ void codesearch_index::dump_metadata() {
     hdr_.name_off = stream_.tellp();
     dump_string(cs_->name());
 
+    hdr_.timestamp_off = stream_.tellp();
+    dump_int64(cs_->index_timestamp());
```

Comment:
Seems to me it would make sense to just store the timestamp in the header, instead of using the offset pattern? For large blobs we store the offsets, but for scalars we can just put a `uint64_t timestamp` directly in the header and it should work fine.